### PR TITLE
fix: humanize coordinator delegation_failure — no raw JSON to principal (#1329)

### PR DESCRIPTION
**Pull request by josephfung**
## Summary

- Closes #1329
- When the top-level coordinator encounters a non-retryable specialist delegation failure, the `_curia_protocol: delegation_failure` JSON signal now never surfaces to the principal. The coordinator emits a human-readable message instead.
- The JSON protocol signal is preserved for specialists (unchanged), whose `delegation_failure` payload is consumed by the parent coordinator via the delegate skill tool result.

## What changed

**`src/agents/runtime.ts`** — Added `isCoordinator?: boolean` to `AgentConfig`. In the `pendingDelegationEscalation` short-circuit, when `isCoordinator` is true the runtime constructs a plain-language message from the failure fields (`agent`, `reason`, `possiblySucceeded`, `escalated`) instead of emitting the JSON protocol payload. Includes a guard for an empty agent name (external payload) and an observability log when the generic else-branch fires for non-timeout/blocked reasons.

**`src/index.ts`** — Passes `isCoordinator: agentConfig.role === 'coordinator'` when constructing each `AgentRuntime`.

**`tests/unit/agents/runtime.test.ts`** — New test asserting that `isCoordinator: true` produces output with no `_curia_protocol` literal and with human-readable references to the specialist name, background completion, and escalation.

## Test plan

- [ ] New test passes: `humanizes delegation failure for the top-level coordinator — no _curia_protocol in output (#1329)`
- [ ] Existing delegation tests still pass (specialist path unchanged — tests without `isCoordinator: true` still assert protocol JSON)
- [ ] Typecheck clean
- [ ] Full test suite passes (4771 tests)

### CODE REVIEW

**Comment by josephfung** [comment]
@coderabbitai review

File: src/agents/runtime.ts [hunk 1]
```diff
@@ -1351,15 +1357,50 @@ export class AgentRuntime {
...
+          } else {
+            // maxTurns, maxConsecutiveErrors, tool_error, api_error, or unknown reason.
+            logger.info(
+              { agentId, conversationId, targetAgent: esc.agent, reason: esc.reason },
+              'Humanizing coordinator delegation failure with generic message',
+            );
+            parts.push(`The ${agentLabel} wasn't able to complete the task.`);
+          }
+          if (esc.escalated) parts.push("I've logged a follow-up task to review the outcome.");
+          escalationContent = parts.join(' ');
```
[the remark this answers is not in the document]
Comment (josephfung) [reply src/agents/runtime.ts]: Fixed in the latest commit. Changed the fallback to bare `'specialist'` (no article) since all three message templates unconditionally prepend "the " — so `'a specialist'` was producing "the a specialist". Also added the three missing test cases for `blocked`, `tool_error` (generic fallback), and empty `esc.agent`.
